### PR TITLE
Fix for #366 to add error handling for missing primary_state attribute

### PR DESCRIPTION
**Pull request by jonlives**
Signed-off-by: Jon Cowie <jonlives@gmail.com>

### Description

Fixes issue #366 to add error handling when primary state attribute is missing

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant.

Please ensure commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/google_kms_crypto_key.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@ def to_s
   end
 
   def primary_state
-    @fetched['primary']['state']
+    @fetched['primary']['state'] unless @fetched.dig('primary', 'state').nil?
```
Comment (sathish-progress) [inline libraries/google_kms_crypto_key.rb:69]: I guess you want to do this 
```suggestion
    @fetched.dig('primary', 'state')
```

Comment (jonlives) [reply libraries/google_kms_crypto_key.rb:69]: I don't believe this does the same - I need to return the value of @fetched['primary']['state'], or return nil if any of the values in that nested hash are nil ( avoiding catching NoMethodError if for example @fetched[;primary'] is nil)

Comment (sathish-progress) [reply libraries/google_kms_crypto_key.rb:69]: yeah it does exactly what you have asked for .. reference -> https://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig

Comment (sathish-progress) [reply libraries/google_kms_crypto_key.rb:69]: Attached the behaviour here
Screenshot 2021-11-22 at 11 21 23 PM

**Review by sathish-progress — APPROVED** [review]
LGTM

**Pull request merged**